bundles/statistics/statsgrid2016/components/RegionsetViewer.js

```diff
@@ -0,0 +1,150 @@
+                        }
+                    });
+                });
+            });
+
+            service.getRegions(currentRegion, function(er, regions){
+                var features = [];
+                regions.forEach(function(region){
+                    features.push(region.geojson);
+                });
+                var geoJSON = {
+                    'type': 'FeatureCollection',
+                    'crs': {
+                        'type': 'name',
+                        'properties': {
+                          'name': sandbox.getMap().getSrsName()
+                        }
+                      },
+                      'features': features
+                };
+                var params = [geoJSON, {
+                    clearPrevious: true,
+                    featureStyle: {
+                        fill: {
+                            color: 'rgba(255,0,0,0.0)'
+                        },
+                        stroke : {
+                            color: '#000000',
+                            width: 1
+                        }
+                    },
+                    optionalStyles: optionalStyles,
+                    layerId: me.LAYER_ID
+                }];
+                sandbox.postRequestByName(
+                    'MapModulePlugin.AddFeaturesToMapRequest',
+                    params
+                );
+            });
+        });
+    },
+/****** PRIVATE METHODS ******/
+    /**
+     * Listen to events that require re-rendering the UI
+     */
+    _bindToEvents : function() {
+        var me = this;
+        var sandbox = me.sb;
+        var state = me.service.getStateService();
+
+        me.service.on('StatsGrid.IndicatorEvent', function(event) {
+            // if indicator is removed/added
+            me.render();
+        });
+
+        me.service.on('StatsGrid.ActiveIndicatorChangedEvent', function(event) {
+            // Always show the active indicator
+            me.render(state.getRegion());
+        });
+
+        me.service.on('StatsGrid.RegionsetChangedEvent', function(event) {
+            // Need to update the map
+            me.render(state.getRegion());
+        });
+
+        me.service.on('StatsGrid.RegionSelectedEvent', function(event){
+            me.render(event.getRegion());
+        });
+
+        me.service.on('StatsGrid.ClassificationChangedEvent', function(event) {
+            // Classification changed, need update map
+            me.render(state.getRegion());
+        });
+
+        me.service.on('FeatureEvent', function(event){
+            if(event.getParams().operation === 'click' && event.hasFeatures()) {
+                // resolve region
+                var features = event.getParams().features[0];
+                var region = features.geojson.features[0].properties.id;
+
+                if(me.instance.conf.scrollDatatable) {
+                    state.setComponentState('datatable', {
```

Comment:
As background info: This was used to detect where the selection originated (grid/map) to prevent scrolling grid if the user clicked on the grid. Let's solve this by replacing "setComponentState"  with adding a "component id" to selectRegion-function and the event it sends out. Could be "triggeredBy" with a value as one of "map", "grid" (and in the future possibly "graph" or "rpc").